cibuildwheel/backend.py

```diff
@@ -0,0 +1,296 @@
+            self.call(
+                "python",
+                "-m",
+                "pip",
+                "wheel",
+                package_dir,
+                f"--wheel-dir={built_wheel_dir}",
+                "--no-deps",
+                *verbosity_flags,
+            )
+        elif build_options.build_frontend == "build":
+            config_setting = " ".join(verbosity_flags)
+            build_env = self.venv.env.copy()
+            if self.venv.constraints_path:
+                # Bug in pip <= 21.1.3 - we can't have a space in the
+                # constraints file, and pip doesn't support drive letters
+                # in uhi. After probably pip 21.2, we can use uri.
+                assert " " not in str(self.venv.constraints_path)
+                build_env["PIP_CONSTRAINT"] = str(self.venv.constraints_path)
+                build_env["VIRTUALENV_PIP"] = str(self.venv.pip_version)
+            self.call(
+                "python",
+                "-m",
+                "build",
+                package_dir,
+                "--wheel",
+                f"--outdir={built_wheel_dir}",
+                f"--config-setting={config_setting}",
+                env=build_env,
+            )
+        else:
+            assert_never(build_options.build_frontend)
+
+        built_wheel = next(self.platform.glob(built_wheel_dir, "*.whl"))
+        if built_wheel.name.endswith("none-any.whl"):
+            raise NonPlatformWheelError()
+        return built_wheel
+
+    def _repair_wheel(
+        self, built_wheel: PurePath, repaired_wheel_dir: PurePath
+    ) -> Sequence[PurePath]:
+        build_options = self.build_options
+        # repair the wheel
+        if build_options.repair_command:
+            log.step("Repairing wheel...")
+            repair_kwargs: Dict[str, PathOrStr] = {
+                "wheel": built_wheel,
+                "dest_dir": repaired_wheel_dir,
+            }
+            self.update_repair_kwargs(repair_kwargs)
+            repair_command_prepared = prepare_command(build_options.repair_command, **repair_kwargs)
+            self.shell(repair_command_prepared)
+        else:
+            self.platform.move_files(built_wheel, dest=repaired_wheel_dir)
+
+        return tuple(self.platform.glob(repaired_wheel_dir, "*.whl"))
+
+    def run(self, repaired_wheel_dir: PurePath) -> Sequence[PurePath]:
+        assert self._can_run
+        assert not self._running
+        self._running = True
+        with self._setup_build_env():
+            self._install_build_tools()
+            self._before_build()
+            with self.platform.tmp_dir(self.identifier + "-built-wheel") as built_wheel_dir:
+                built_wheel = self._build_wheel(built_wheel_dir)
+                wheels = self._repair_wheel(built_wheel, repaired_wheel_dir)
+            # save build environment
+            self._constraints_dict = self.venv.constraints_dict
+        return wheels
+
+
+def test_one(
```

Comment:
Starting with `test_` makes this look like a pytest test.

Reply:
changed to `do_test_wheel`